File: scala/src/main/scala/model/Item.scala
package model

class Item {

  private var product: Product = new Product
  private var cost = .0
  private var tax = .0
  private var shipping = .0

  def constructor(newProduct: Product, newCost: Double, newTax: Double, newShipping: Double): Unit = {
      product = newProduct
      cost = newCost
      tax = newTax
      shipping = newShipping
  }

  def setProduct(newProduct: Product): Unit = {
     product = newProduct
  }
  def getProduct: Product = product

  def setCost(newCost: Double): Unit = {
    cost = newCost
  }
  def getCost: Double = cost

  def setTax(newTax: Double): Unit = {
    tax = newTax
  }
  def getTax: Double = tax

  def setShipping(newShipping: Double): Unit = {
    shipping = newShipping
  }
  def getShipping: Double = shipping

}

File: java/src/main/model/Item.java
package main.model;

public class Item {

    private Product produto;
    private double cost;
    private double tax;
    private double shipping;

    public Item(Product produto, double cost, double tax, double shipping) {
        this.produto = produto;
        this.cost = cost;
        this.tax = tax;
        this.shipping = shipping;
    }

    public Product getProduto() {
        return produto;
    }

    public void setProduto(Product produto) {
        this.produto = produto;
    }

    public double getCost() {
        return cost;
    }

    public void setCost(double cost) {
        this.cost = cost;
    }

    public double getTax() {
        return tax;
    }

    public void setTax(double tax) {
        this.tax = tax;
    }

    public double getShipping() {
        return shipping;
    }

    public void setShipping(double shipping) {
        this.shipping = shipping;
    }
}
